compendium/ConfigurationAdmin/test/TestConfigAdmin.cpp

```diff
@@ -843,3 +849,100 @@ TEST_F(ConfigAdminTests, testMultipleManagedFactoriesInBundleGetUpdate)
                             1);
                 });
 }
+// <summary>
+/// Used by ConfigAdminTests.testManagedServiceRemoveConfigurationsDeadlock
+/// </summary>
+namespace cppmicroservices {
+namespace test {
+class TestManagedServiceInterface2
+  : public cppmicroservices::service::cm::ManagedService
+{
+public:
+  virtual ~TestManagedServiceInterface2() noexcept = default;
+  void Updated(const cppmicroservices::AnyMap&) override = 0; 
+};
+
+class TestManagedServiceImpl : public TestManagedServiceInterface2
+{
+public:
+  TestManagedServiceImpl(cppmicroservices::Framework& framework)
+    : m_framework(framework)
+   {}
+  virtual ~TestManagedServiceImpl() noexcept = default;
+
+  void Updated(const cppmicroservices::AnyMap& props) override
+  { 
+    // The Updated method is called for both Updated and Removed operations. 
+    // For the Removed operations, the properties are empty.
+    if (props.empty()) {
+      auto ctx = m_framework.GetBundleContext();
+      auto bundles = ctx.GetBundles();
+      for (auto& b : bundles) {
+        b.Stop();
```

Comment:
Only the `TestBundleManagedServiceDeadlock` bundle should be stopped. This will stop all bundles installed and started by the framework.